cpp/src/experimental/louvain.cuh

```diff
@@ -418,11 +421,10 @@ class Louvain {
       base_src_vertex_id_ = graph_view.get_local_adj_matrix_partition_row_first(0);
       base_dst_vertex_id_ = graph_view.get_local_adj_matrix_partition_col_first(0);
 
-      raft::copy(&local_num_edges_,
-                 graph_view.offsets() + graph_view.get_local_adj_matrix_partition_row_last(0) -
-                   graph_view.get_local_adj_matrix_partition_row_first(0),
-                 1,
-                 stream_);
+      local_num_edges_ = edge_t{0};
+      for (size_t i = 0; i < graph_view.get_number_of_local_adj_matrix_partitions(); ++i) {
```

Comment:
Consider replacing custom loop by algorithm (in this case, probably `std::accumulate()`).

Reply:
This is a temporary solution until @seunghwak exposes a `get_number_of_local_edges()` method on the graph view.

I cannot see a clean way - given the current `graph_view` API - to simplify this into an std algorithm.  The only good way I saw was to use `std::transform_reduce` which isn't available in C++ 14 (our current baseline requirement).  I could use `thrust::transform_reduce` and have it specify a host function, I suppose.  I'm inclined to let Seunghwa add the method (probably in version 0.19) and using it here rather than create a sophisticated temporary solution.

Reply:
```
auto sz = graph_view.get_number_of_local_adj_matrix_partitions();
local_num_edges_ = thrust::reduce(thrust::host, 
                                                         thrust::make_counting_iterator(0),
                                                         thrust::make_counting_iterator(sz),
                                                         [graph_view] (auto indx1, auto indx2){
                                                            return graph_view.get_number_of_local_adj_matrix_partition_edges(indx1)+
graph_view.get_number_of_local_adj_matrix_partition_edges(indx2);
                                                         });
```
Wouldn't that work?

Reply:
I don't think that works.  The reduction function has to be able to be applied to the output of the reduction function, it can't do a transformation from indices to sums.  That's why transform_reduce exists.

I just pushed a new version using `thrust::transform_reduce`.  It's a bit ugly because the graph_view object is not trivially copyable.  Ultimately this will go away once the new method is provided.